beets/ui/commands.py

```diff
@@ -489,6 +494,13 @@ def choose_candidate(candidates, singleton, rec, cur_artist=None,
         assert cur_artist is not None
         assert cur_album is not None
 
+    # Build helper variables for extra choices.
+    # TODO: move "abort" choice always to the end?
+    # TODO: add plugin name info on choices ("(mb) Print tracks")?
+    extra_opts = tuple([c.long for c in extra_choices])
+    extra_actions = {c.long.strip(lowercase + whitespace)[0].lower(): c.id
```

Comment:
Great! I was holding off the writing documentation until making sure the changes were on the right track, but will add it soon. On a second look, I will probably go for the explicit route - it should be rather lightweight for the plugin writers, and will probably make things a bit more clear and less prone to surprises.